provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -211,21 +198,23 @@ static int generate_root_ca_info(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNA
             result = 0;
         }
     }
+    ecc_signature_destroy(&tbs_sig);
     return result;
 }
 
-static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNATURE tbs_sig, CERTIFICATE_SIGNING_TYPE signing_type)
+static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, CERTIFICATE_SIGNING_TYPE signing_type)
 {
     int result;
     uint8_t der_buffer[DER_MAX_TBS] = { 0 };
     DERBuilderContext der_ctx = { 0 };
     RIOT_STATUS status;
+    RIOT_ECC_SIGNATURE tbs_sig = { 0 };
 
     if (signing_type == type_self_sign)
     {
         // Build the TBS (to be signed) region of DeviceID Certificate
         DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub) != 0)
+        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub, (uint8_t*)&riot_info->device_id_pub, sizeof(riot_info->device_id_pub)) != 0)
```

Comment:
Please explain why we need to pass the last two parameters here and everywhere else with `X509GetDeviceCertTBS`.

Reply:
API change, they are optional parameters, I wasn't fully seeing them do anything whilst in the debugging sessions that relates to our code but need them to be able to compile, 0 and NULL are non valid pass ins since they get checked in the parameter checking of the API

Reply:
The optional parameters were added to identify the public key used to verify the signature on the DeviceID cert.  See the Authority Key Identifier extension (OID 2.5.29.35).  This is not strictly necessary but is included by convention.  There is no requirement within the RIoTCrypt API for these parameters, if RootKeyPub is NULL both parameters are ignored.

Reply:
For this particular call, since the DeviceID is self-signed, I don't think there is an expectation that this extension will be included in the cert (i.e., the two optional parameters should be NULL,0).  There's no harm in it, but it doesn't tell us anything we don't already know.

Reply:
Fixed